### PR TITLE
Autodetect service

### PR DESCRIPTION
**Pull request by lewisporter**
Adds rakefile, autodetects service name from env variables, updates dependencies, populates request id header that HAproxy uses

### CODE REVIEW

File: request_tracing.go [hunk 1]
```diff
@@ -39,7 +46,8 @@ func TracingMiddleware(h http.HandlerFunc) http.HandlerFunc {
 			defer func() {
 				if requestID := requestWithContext.Context().Value(ctxRequestIDKey); requestID != nil {
 					if requestIDStr, ok := requestID.(string); ok {
-						w.Header().Set(headerRequestIDKey, requestIDStr)
+						setRequestID(w, headerRequestIDKey, requestIDStr)
```
Comment (dsouzajude) [inline request_tracing.go]: What would be the difference b/w `headerRequestIDKey` and `headerUniqueIDKey `?

Comment (dsouzajude) [inline request_tracing.go]: Can we use one of these instead?

File: request_tracing.go [hunk 2]
```diff
@@ -7,27 +7,34 @@ import (
...
 	ctxRequestIDKey    ctxRequestIDKeyType = "request-id"
 	headerRequestIDKey string              = "X-Request-ID"
+	headerUniqueIDKey  string              = "X-Unique-ID"
 )
 
+var userAgent string
+
+// init reads the 'SERVICE_NAME' environment variable so that if it is not set manually, we have something meaningful
+func init() {
+	userAgent = os.Getenv(`SERVICE_NAME`)
```
Comment (dsouzajude) [inline request_tracing.go]: Nice !!!

File: request_tracing.go (hunk 1 above)
Comment (lewisporter) [inline request_tracing.go]: Yep! For now, I want to leave it to ensure compatibility, but probably we can remove the `x-request-id` header

**Comment by dsouzajude** [comment]
@lewisporter How do i know this works? Do you have tests for these changes?

**Comment by lewisporter** [comment]
@dsouzajude These changes will be covered by existing tests.

File: request_tracing.go (hunk 1 above)
Comment (dsouzajude) [inline request_tracing.go]: Fair enough.

**Pull request merged**